test/gateways/authorize_net_test.exs

```diff
@@ -23,26 +26,41 @@ defmodule Gringotts.Gateways.AuthorizeNetTest do
   @bad_amount "a"
 
   @opts [
-    config: %{name: "64jKa6NA", transactionKey: "4vmE338dQmAN6m7B"}, 
-    refId: "123456", 
+    config: %{name: "64jKa6NA", transactionKey: "4vmE338dQmAN6m7B"},
+    refId: "123456",
     order: %{invoiceNumber: "INV-12345", description: "Product Description"}, 
     lineItem: %{itemId: "1", name: "vase", description: "Cannes logo", quantity: "18", unitPrice: "45.00" }
   ]
 
   describe "purchase" do
     test "test_successful_purchase" do
-      {:ok, response} = ANet.purchase(@amount, @card, @opts)
-      assert response.raw["createTransactionResponse"]["messages"]["resultCode"] == "Ok"
+      with_mock HTTPoison, 
+        [request: fn(_method, _url, _body, _headers) -> MockResponse.successful_purchase_response end] do
```

Comment:
Oops! Please confirm that the request is getting mocked. I ran your tests and:
1. They are all extremely slow (above 200ms).
2. "test_successful_purchase" succeeds the first time, but fails on subsequent runs. [Here's the stacktrace](https://pastebin.com/AY4s5UP7).

It looks like the requests are hitting the ANet server.

Reply:
sorry I don't agree with this I don't know what is causing it to behave like this, just ran it to confirm 
```elixir
Gringotts.Gateways.AuthorizeNetTest
  * test purchase test_successful_purchase (57.8ms)
  * test purchase test_bad_amount_purchase (25.9ms)
  * test purchase test_bad_card_purchase (31.4ms)
  * test purchase test network error (30.3ms)```

Reply:
And I ran them repeatedly just to check if it is hitting the gateway. Had they been hitting the server I would have got a failed response as my gateway fails subsequent requests in a given amount of time.

Reply:
I think that is because in my last pull request though I had added tests, but they were directly hitting the gateway, I had not used mock there. I have added mock in this pull request.

Reply:
I apologise, I read code on Github, but did not pull your commits (so the tests were stale). My bad 😓